src/portals/hope/features/tools/template-messages/use-template-messages.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { createContext, useContext, useState } from 'react'
+import toast from 'react-hot-toast'
+import { useInsecurePersistentState } from '@hedvig-ui/hooks/use-insecure-persistent-state'
+import { TemplateMessages } from './components/TemplateMessages'
+
+export enum Markets {
+  Sweden = 'SWEDEN',
+  Denmark = 'DENMARK',
+  Norway = 'NORWAY',
+}
+
+export interface Message {
+  market: Markets
+  text: string
+}
+
+export interface TemplateMessage {
+  name: string
+  id: string
+  markets: Markets[]
+  messages: Message[]
+  messageEn: string
+  withExpiry?: boolean
+  expiryDate?: string | null
+  pinned?: boolean
+}
+
+interface TemplateMessagesContextProps {
+  templates: TemplateMessage[]
+  show: () => void
+  createTemplate: (template: TemplateMessage) => void
+  editTemplate: (template: TemplateMessage) => void
+  deleteTemplate: (id: string) => void
+  pinTemplate: (id: string) => void
+  select: (text: string) => void
+  selected: string | null
+  currentMarket: Markets
+  changeCurrentMarket: (market: Markets) => void
```

Comment:
Maybe have a more consistent naming here, something like

```suggestion
  templates: TemplateMessage[]
  show: () => void
  create: (template: TemplateMessage) => void
  edit: (template: TemplateMessage) => void
  delete: (id: string) => void
  pin: (id: string) => void
  select: (text: string) => void
  selected: string | null
  market: Market
  setMarket: (market: Market) => void
```

I'm thinking that the hook implies some sort of namespacing when you're using it? I guess you also could go the other way around, suffixing everything with *Template